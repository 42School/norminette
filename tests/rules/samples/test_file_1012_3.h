/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   envvar.h                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: abaur <abaur@student.42.fr>                +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2020/10/07 13:51:00 by abaur             #+#    #+#             */
/*   Updated: 2020/10/07 13:51:00 by abaur            ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#ifndef TEST_FILE_1012_3_H
# define TEST_FILE_1012_3_H

# include "../dynarray/dynarray.h"

t_dynarray	g_envarray;

/*
** Duplicates the provided array to initialize the environnement.
** @param char** environ	An array of strings that will be used as environnem
** ent variables. These are not checked for invalid syntax.
** @param char**	The resulting environnement, or NULL in case of error.
*/

char		**envvarinit(char **environ);

/*
** Frees all the internal pointers of the environnement.
*/

void		envvardeinit(void);

/*
** Fetches the value of an environnement variable.
** @param const char* name	The name of the variable.
** @return char*	An allocated copy of the variable's value.
*/

char		*get_env_var(const char *name);

/*
** Sets an environnement variable.
** @param char* value 	A string formated as "name=value".
** 	This string is NOT checked for invalid syntax.
** 	This exact pointer is stored internally, and should notbe modified or freed
**  afterward.
** @return bool
** 	true 	OK;
** 	false 	Error;
*/

short		set_env_var_raw(char *value);

/*
** Sets an environnement variable.
** @param const char* name	The name of the variable to set.
** 	This name is NOT checked for invalid caracters, except for '='.
** @param const char* value	The value to set.
** return bool
** 	true 	OK
** 	false	Error
*/

short		set_env_var(const char *name, const char *value);

/*
** Checks the validity of an environnement variable's name.
** The special name '?' is considered valid.
** @param const char* name	The name to validate
** @return char*	A pointer to the first invalid character, else a pointer to
**  the null terminator.
*/

char		*validate_var_name(const char *name);

#endif