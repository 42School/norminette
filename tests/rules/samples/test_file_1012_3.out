[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 1":
		<MULT_COMMENT=/* ************************************************************************** */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 2":
		<MULT_COMMENT=/*                                                                            */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 3":
		<MULT_COMMENT=/*                                                        :::      ::::::::   */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 4":
		<MULT_COMMENT=/*   envvar.h                                           :+:      :+:    :+:   */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 5":
		<MULT_COMMENT=/*                                                    +:+ +:+         +:+     */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 6":
		<MULT_COMMENT=/*   By: abaur <abaur@student.42.fr>                +#+  +:+       +#+        */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 7":
		<MULT_COMMENT=/*                                                +#+#+#+#+#+   +#+           */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 8":
		<MULT_COMMENT=/*   Created: 2020/10/07 13:51:00 by abaur             #+#    #+#             */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 9":
		<MULT_COMMENT=/*   Updated: 2020/10/07 13:51:00 by abaur            ###   ########.fr       */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 10":
		<MULT_COMMENT=/*                                                                            */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 11":
		<MULT_COMMENT=/* ************************************************************************** */> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 12":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 13":
		<IFNDEF=#ifndef TEST_FILE_1012_3_H> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 14":
		<DEFINE=# define TEST_FILE_1012_3_H> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 15":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 16":
		<INCLUDE=# include "../dynarray/dynarray.h"> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 17":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsVarDeclaration[0m In "GlobalScope" from "None" line 18":
		<IDENTIFIER=t_dynarray> <TAB> <IDENTIFIER=g_envarray> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 19":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 20":
		<MULT_COMMENT=/*
** Duplicates the provided array to initialize the environnement.
** @param char** environ	An array of strings that will be used as environnem
** ent variables. These are not checked for invalid syntax.
** @param char**	The resulting environnement, or NULL in case of error.
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 26":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 27":
		<CHAR> <TAB> <TAB> <MULT> <MULT> <IDENTIFIER=envvarinit> <LPARENTHESIS> <CHAR> <SPACE> <MULT> <MULT> <IDENTIFIER=environ> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 28":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 29":
		<MULT_COMMENT=/*
** Frees all the internal pointers of the environnement.
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 32":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 33":
		<VOID> <TAB> <TAB> <IDENTIFIER=envvardeinit> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 34":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 35":
		<MULT_COMMENT=/*
** Fetches the value of an environnement variable.
** @param const char* name	The name of the variable.
** @return char*	An allocated copy of the variable's value.
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 40":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 41":
		<CHAR> <TAB> <TAB> <MULT> <IDENTIFIER=get_env_var> <LPARENTHESIS> <CONST> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=name> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 42":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 43":
		<MULT_COMMENT=/*
** Sets an environnement variable.
** @param char* value 	A string formated as "name=value".
** 	This string is NOT checked for invalid syntax.
** 	This exact pointer is stored internally, and should notbe modified or freed
**  afterward.
** @return bool
** 	true 	OK;
** 	false 	Error;
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 53":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 54":
		<SHORT> <TAB> <TAB> <IDENTIFIER=set_env_var_raw> <LPARENTHESIS> <CHAR> <SPACE> <MULT> <IDENTIFIER=value> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 55":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 56":
		<MULT_COMMENT=/*
** Sets an environnement variable.
** @param const char* name	The name of the variable to set.
** 	This name is NOT checked for invalid caracters, except for '='.
** @param const char* value	The value to set.
** return bool
** 	true 	OK
** 	false	Error
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 65":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 66":
		<SHORT> <TAB> <TAB> <IDENTIFIER=set_env_var> <LPARENTHESIS> <CONST> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=name> <COMMA> <SPACE> <CONST> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=value> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 67":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsComment[0m In "GlobalScope" from "None" line 68":
		<MULT_COMMENT=/*
** Checks the validity of an environnement variable's name.
** The special name '?' is considered valid.
** @param const char* name	The name to validate
** @return char*	A pointer to the first invalid character, else a pointer to
**  the null terminator.
*/> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 75":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 76":
		<CHAR> <TAB> <TAB> <MULT> <IDENTIFIER=validate_var_name> <LPARENTHESIS> <CONST> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=name> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 77":
		<NEWLINE>
[36mtest_file_1012_3.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 78":
		<ENDIF=#endif> 
test_file_1012_3.h: OK!
